### PR TITLE
[CN-437] Fix for HotBackup with BucketURI always triggering backup upload

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**
HotBackup with BucketURI could trigger backup upload for Local backupType which is not expected as there wont be sidecar upload agent present.

### CODE REVIEW

File: controllers/hazelcast/hot_backup_status.go [hunk 1]
```diff
@@ -56,7 +56,12 @@ func hotBackupState(hbs HotRestartState, currentState hazelcastv1alpha1.HotBacku
 		return hazelcastv1alpha1.HotBackupFailure
 	case "SUCCESS":
 		if currentState == hazelcastv1alpha1.HotBackupUnknown {
-			switch backupType {
+			// quick exit if no persistance is set
```
[the remark this answers is not in the document]
Comment (dzeromski-hazelcast) [reply controllers/hazelcast/hot_backup_status.go:59]: You mean hz.Spec.Persistence check? Or case "SUCCESS"? Please remember that is hbs.BackupTaskState which will be in SUCCESS state and that is expected.

[the remark this answers is not in the document]
Comment (dzeromski-hazelcast) [reply controllers/hazelcast/hot_backup_status.go:59]: Probably not now but it is good to check for nil. Better safe than sorry :)

**Review by hasancelik — APPROVED** [review]
LGTM 👍 After resolve @mtyazici's latest comment(https://github.com/hazelcast/hazelcast-platform-operator/pull/261#discussion_r908594081).

File: controllers/hazelcast/hot_backup_controller.go [hunk 2]
```diff
@@ -139,10 +134,10 @@ func (r *HotBackupReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 			return updateHotBackupStatus(ctx, r.Client, hb, failedHbStatus(err))
 		}
 
-		r.reconcileHotBackupStatus(ctx, hb, backupType)
+		r.reconcileHotBackupStatus(ctx, hb, h)
 	}
 
-	if hb.Spec.BucketURI != "" {
+	if h.Spec.Persistence != nil && h.Spec.Persistence.BackupType == hazelcastv1alpha1.External {
```
[the remark this answers is not in the document]
Comment (dzeromski-hazelcast) [reply controllers/hazelcast/hot_backup_controller.go]: Switched to hz.Spec.Persistence.IsExternal()

**Pull request merged**